src/libcadet/model/parts/ConvectionDispersionOperator.cpp

```diff
@@ -185,29 +186,21 @@ bool ConvectionDispersionOperatorBase::configure(UnitOpIdx unitOpIdx, IParameter
  */
 bool ConvectionDispersionOperatorBase::notifyDiscontinuousSectionTransition(double t, unsigned int secIdx)
 {
-	double prevVelocity = static_cast<double>(_curVelocity);
-
 	// If we don't have cross section area, velocity is given by parameter
+	double _dir_old = _dir;
 	if (_crossSection <= 0.0)
-		_curVelocity = getSectionDependentScalar(_velocity, secIdx);
+		_dir = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
```

Comment:
What happens if `_velocity` is not just one of `{-1.0, 1.0}`? I suggest to normalize (i.e., only assign `-1.0` or `1.0` to `_dir`) just to be on the safe side.

Reply:
Good point! I will simply check the sign of the velocity that we read (also according to the [documentation](https://cadet.github.io/master/modelling/unit_operations/general_rate_model.html#specification-of-flow-rate-velocity-and-direction). Should I also change `dir` to an int type?

Reply:
Doesn't make a real difference. So it's your choice.